### PR TITLE
rbd: discard flattening error if the image doesnt have parent

### PR DESCRIPTION
**Pull request by Madhu-1**
if the image is created without flattening image-feature the image will get few image-features by default, deep-flatten
is one of them. if the image doesn't have any parent the rbd image flattening will fail, This commit discards error message if the image doesn't have any parent.

more  info https://github.com/ceph/ceph-csi/issues/1217
Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

### CODE REVIEW

**Comment by Madhu-1** [comment]
if we don't want to discard error. we have to default the layering during image creation time. @dillaman any suggestion here?

File: internal/rbd/rbd_util.go [hunk 1]
```diff
@@ -478,6 +479,25 @@ func (rv *rbdVolume) flattenRbdImage(ctx context.Context, cr *util.Credentials,
...
+func (rv *rbdVolume) flatten() error {
+	rbdImage, err := rv.open()
+	if err != nil {
+		return err
+	}
+	defer rbdImage.Close()
+
+	err = rbdImage.Flatten()
+	if err != nil {
+		// go-ceph returns `rbd: ret=22, Invalid argument` if the
```
Comment (nixpanic) [inline internal/rbd/rbd_util.go:491]: `Invalid argument` is not a unique error for `no parent`. You should probably check with `rbdImage.GetParentInfo()` and see if the name/snapname is stored in the `[]byte` that is passed to the function.

**Comment by Madhu-1** [comment]
@MergifyIO rebase

File: internal/rbd/rbd_util.go (hunk 1 above)
Comment (nixpanic) [reply internal/rbd/rbd_util.go:491]: this is still a TODO

**Comment by Madhu-1** [comment]
@nixpanic I didn't fix the comment earlier, I was just testing the centos CI. fixed it now. PTAL

File: internal/rbd/rbd_util.go [hunk 2]
```diff
@@ -478,6 +479,44 @@ func (rv *rbdVolume) flattenRbdImage(ctx context.Context, cr *util.Credentials,
...
+func (rv *rbdVolume) flatten() error {
+	rbdImage, err := rv.open()
+	if err != nil {
+		return err
+	}
+	defer rbdImage.Close()
+
+	err = rbdImage.Flatten()
+	if err != nil {
+		parent, pErr := rv.getParentName()
```
Comment (nixpanic) [inline internal/rbd/rbd_util.go:509]: This looks good, although some comments would be in order to make it easier to understand the intended "flatten() returns no error if the image does not have a parent".

**Comment by nixpanic** [comment]
Needs to be rebased to pull in the `scripts/install-helm.sh` fixes for CI.

File: internal/rbd/rbd_util.go (hunk 2 above)
Comment (Madhu-1) [reply internal/rbd/rbd_util.go:510]: added a comment.

**Comment by Madhu-1** [comment]
@humblec PTAL

**Comment by nixpanic** [comment]
@mergifyio rebase

**Comment by nixpanic** [comment]
This needs to be rebased on top of #1348 so that `/test commitlint` can work correctly.

**Comment by Madhu-1** [comment]
@nixpanic PTAL

**Comment by Madhu-1** [comment]
@MergifyIO refresh

**Comment by Madhu-1** [comment]
@mergifyIO rebase

**Comment by Madhu-1** [comment]
commitlint bot was failing due to rebase https://jenkins-ceph-csi.apps.ocp.ci.centos.org/job/commitlint/23/console

**Comment by Madhu-1** [comment]
@MergifyIO rebase

**Comment by humblec** [comment]
@mergify rebase

**Comment by humblec** [comment]
@mergify rebase

**Comment by nixpanic** [comment]
This can be rebased (again) after Mergify updates the master branch. That way, the GitHub commitlint App should pick it up again :crossed_fingers:

**Comment by Madhu-1** [comment]
@mergifyio rebase

**Pull request merged**